rotors_gazebo_plugins/CMakeLists.txt

```diff
@@ -1,142 +1,422 @@
+  if (NOT NO_ROS)
+    add_dependencies(gazebo_geotagged_images_plugin ${catkin_EXPORTED_TARGETS})
+  endif()
+  list(APPEND targets_to_install gazebo_geotagged_images_plugin)
+else()
+  message(STATUS "Gazebo version is less than 5, not building gazebo_geotagged_images_plugin.cpp.")
+endif()
+
+#========================================= GPS PLUGIN ===========================================//
+add_library(rotors_gazebo_gps_plugin SHARED src/gazebo_gps_plugin.cpp)
+target_link_libraries(rotors_gazebo_gps_plugin ${catkin_LIBRARIES} ${GAZEBO_LIBRARIES})
+if (NOT NO_ROS)
+  add_dependencies(rotors_gazebo_gps_plugin ${catkin_EXPORTED_TARGETS})
+endif()
+list(APPEND targets_to_install rotors_gazebo_gps_plugin)
+
+#========================================= IMU PLUGIN ===========================================//
+add_library(rotors_gazebo_imu_plugin SHARED src/gazebo_imu_plugin.cpp)
+target_link_libraries(rotors_gazebo_imu_plugin ${catkin_LIBRARIES} ${GAZEBO_LIBRARIES})
+if (NOT NO_ROS)
+  add_dependencies(rotors_gazebo_imu_plugin ${catkin_EXPORTED_TARGETS})
+endif()
+list(APPEND targets_to_install rotors_gazebo_imu_plugin)
+
+#======================================== LIDAR PLUGIN ==========================================//
+if(${gazebo_VERSION_MAJOR} GREATER 4)
+  add_library(rotors_gazebo_lidar_plugin SHARED src/gazebo_lidar_plugin.cpp)
+  target_link_libraries(rotors_gazebo_lidar_plugin ${catkin_LIBRARIES} ${GAZEBO_LIBRARIES})
+  if (NOT NO_ROS)
+    add_dependencies(rotors_gazebo_lidar_plugin ${catkin_EXPORTED_TARGETS})
+  endif()
+  list(APPEND targets_to_install rotors_gazebo_lidar_plugin)
+else()
+  message(STATUS "Gazebo version is less than 5, not building gazebo_lidar_plugin.cpp.")
+endif()
+
+#===================================== MAGNETOMETER PLUGIN ======================================//
+add_library(rotors_gazebo_magnetometer_plugin SHARED src/gazebo_magnetometer_plugin.cpp)
+target_link_libraries(rotors_gazebo_magnetometer_plugin ${catkin_LIBRARIES} ${GAZEBO_LIBRARIES})
+if (NOT NO_ROS)
+  add_dependencies(rotors_gazebo_magnetometer_plugin ${catkin_EXPORTED_TARGETS})
+endif()
+list(APPEND targets_to_install rotors_gazebo_magnetometer_plugin)
+
+#================================= MAVLINK INTERFACE PLUGIN =====================================//
+
+if(BUILD_MAVLINK_INTERFACE_PLUGIN)
+  message(STATUS "BUILD_MAVLINK_INTERFACE_PLUGIN = TRUE, adding mavros dependency and building mavlink_interface_plugin.")
+else ()
+  message(STATUS "BUILD_MAVLINK_INTERFACE_PLUGIN = FALSE, NOT adding mavros dependency and NOT building mavlink_interface_plugin.")
+endif ()
+
+if (BUILD_MAVLINK_INTERFACE_PLUGIN)
+
+  # We need the MAVLink headers.
+  set(MAVLINK_HEADERS_FOUND FALSE)
+  
+  # First, check to see if MAVLink headers were passed in as variable
+  if(NOT ${MAVLINK_HEADER_DIR} STREQUAL "")
+  
+    message(STATUS "MAVLINK_HEADER_DIR provided as '${MAVLINK_HEADER_DIR}'.")
```

Comment:
Yes that would be tidier, as it would separate concerns. Can we leave for now though in the interests of getting things merged? Can make issue if you want.